### DISCUSSION TITLE
Is a (Semantic) Data Product a Catalog or a Dataset (or both or something else)?

### DISCUSSION THREAD

**Discussion by jgeluk** [EKGF/dprod#11] [General]
In the Data Mesh world, it seems to me that a Data Product is a piece of code that is deployed somewhere serving data.

Which is traditionally called many other things such as "a server", "a database server", "a data service" and so forth.

In an Enterprise Knowledge Graph (EKG) context though, we need something that represents the logical version (or even virtual version) of a Data Product. Something that humans and AIs can reason about, can buy, consume etc.

Should we see such a Data Product as:

a) a "Catalog (or collection) of Datasets"?
b) a "Catalog of (other) Data Products"?
c) an extended (marketable, deployable) version of a Dataset?

**Comment by andrea-gioia** [comment]
The concept of data product can be associated with existing concepts in DCAT. In particular, the following are the possible mappings:

1. **Data Product => Resource:** A data product can be seen as a specific type of catalogable resource, just like a dataset and a data service.
1. **Data Product => Dataset:** A data product can be seen as a specific type of dataset. However, this association is limiting because, in the data mesh approach, a data product is a richer entity than just the exposed data. Additionally, a data product may expose multiple datasets.
1. **Data Product => Catalog:** A data product can be seen as a catalog that exposes the resources it is composed of. 

IMHO the 3rd is the preferable option as shown in [this image](https://app.diagrams.net/#G1SWlRLX3r_MOM1JQn6JzT3IS5MsMUtqis#%7B%22pageId%22%3A%22rtZldOMUc_RTTjouAffK%22%7D). Initially, a data product can be seen as a catalog of its datasets and data services, and in the future, even its internal components (infrastructure and applications).

**Reply by jgeluk** [reply]
The image requires a login which didn't work from my machine (tried two different browsers).

**Reply by jgeluk** [reply]
That makes sense from the perspective of the Data Mesh world where a Data Product is primarily seen as a deployable piece of code serving data managed by one team.

But if you see a Data Product as "something that a data consumer wants to buy" then it makes less sense to treat the Data Product as a DCAT Catalog of Datasets/resources. Because those Datasets are the things that the consumer wants to buy individually. So each of those Datasets becomes a product in itself, so then it would make more sense to see a DataProduct as a sellable/marketable version of a Dataset.

**Reply by jgeluk** [reply]
In your vision of a DataProduct being a Catalog, would all Datasets have the same specifications when it comes to entitlements, applicable policies & regulations, pricing, legal ownership & control details etc? Or should we then extend the DCAT Dataset with all of that? What if a given Dataset is deployed by multiple departments as part of different Data Products?

**Reply by andrea-gioia** [reply]
However, if we identify a data product as a sales unit to which the same conditions (policies, pricing, etc.) apply, then a dataset is not suitable as a mapping. In fact, a dataset has different distributions served by various data services. Therefore, it would be more accurate to associate a data product with a data service that is actually consumed by consumers with specific usage conditions. I'm not entirely fond of this approach though. I would go beyond the perspective of a data product as a purchasing unit. What do you think?

**Reply by andrea-gioia** [reply]
Why not see a data product as a collection of data exposed by a single producer instead? Even a software product is ultimately composed of multiple services, each with its own service agreements. Consumers may only consume some of the services within the software product, but that doesn't make it any less of a product. What's different for a data product?

**Reply by jgeluk** [reply]
I agree! If we want to make this as easy to understand as possible, for everyone, then we should stay as close to the general connotation that everyone has about what a product is in economic terms. Something provided by one supplier and consumed by one consumer, via an agreement in the form of a contract, in exchange for money. 

One data product may however be served in many different ways:

1) By one data-service
2) By multiple data-services
3) By multiple different protocols (JDBC, REST, GraphQL, Kafka, whatever, even FTP) where each of these protocols could be one "output port" on a given host, collection of hosts, or even many services (access points) across the world...
4) It could be that any of those data-services or access points *also* serves any other number of data products.

**Reply by jgeluk** [reply]
I'd even say that when anyone accesses a data product through whatever port or access point, that in the given protocol, whatever that may be, it has to be clear, via a header or token or so, what the corresponding agreement/contract/context is in which the data product is being consumed/accessed.

**Reply by jgeluk** [reply]
For any given data-product we have tons of "detail" that people should be able to specify, which may be referred to in data contracts (and should be stable for the duration of the contract, blockchain comes to mind, and/or versioning of metadata).

Such as:

- Entitlement & Usage policies (bloomberg restricting access to a market data feed to just a number of use cases in a number of departments for instance)
- IP policy
- Usage policies
- Pricing policies
- Jurisdiction specific regulatory and legal policies
- Many others

Basically all of that screams for "graph", or actually "semantic graph", quite complex metadata that all combined would allow for highly automated robust enterprise-grade data services that can serve *any* data product as long as it has all this information onboard.

**Reply by MikeHypercube** [reply]
Isn't the data-set playing the role of a product here?

Specifically it is the 'thing of value' (the economic resource) that is the subject of one side of the contract (the reciprocal commitment from the other party being a commitment to pay for that product under whatever terms e.g. one-off, subscription etc.)
